features/dapp_develop/nodejs.go

```diff
@@ -0,0 +1,142 @@
+
+		txn.sign(SorobanClient.Keypair.fromSecret("{{js .secretKey}}"));
+		let response = await server.sendTransaction(txn);
+		let i = 0;
+		while (response.status === "pending") {
+			i += 1;
+			if (i > 50) {
+				throw new Error("Transaction timed out");
+			}
+			await new Promise(resolve => setTimeout(resolve, 100));
+			response = await server.getTransactionStatus(response.id);
+			switch (response.status) {
+			case "pending": {
+				// noop
+				break;
+			}
+			case "success": {
+				// parse and print the response (assuming it is a vec)
+				// TODO: Move this scval serializing stuff to stellar-base
+				const result = xdr.TransactionResult.fromXDR(response.resultXdr, "base64");
+				const scval = result.result().results()[0].tr().invokeHostFunctionResult().success();
+
+				// Hacky result parsing. We should have some helpers from the
+				// js-stellar-base, or the generated Typescript bindings. But we don't yet as
+				// I'm writing this.
+				let parsed = null;
+				switch (scval.switch()) {
+				case xdr.ScValType.scvU32(): {
+					parsed = scval.u32();
+					break;
+				}
+				case xdr.ScValType.scvI32(): {
+					parsed = scval.i32();
+					break;
+				}
+				case xdr.ScValType.scvObject(): {
+					// Total hack, we just assume the object is a vec. Good enough for now.
+					parsed = scval.obj().vec().map(v => v.sym().toString());
+					break;
+				}
+				default:
+					throw new Error("Unexpected scval type:", scval.switch().name);
+				}
+				console.log(JSON.stringify(parsed));
+				return;
+			}
+			case "error": {
+				throw new Error("Transaction failed:", response);
+			}
+			default:
+				throw new Error("Unknown transaction status: " + response.status);
+			}
+		}
+		throw new Error("Transaction failed:", response);
+	})();
+`
+
+// return the fn response as a serialized string
+// uses secret-key and network-passphrase directly on command
+func invokeContractFromNodeJSTool(deployedContractId, contractName, functionName, param1 string, e2eConfig *e2e.E2EConfig) (string, error) {
+	stdin := &bytes.Buffer{}
+	tmpl, err := template.New("script").Parse(script)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse javascript template")
+	}
+	err = tmpl.Execute(stdin, map[string]string{
+		"contractId":        deployedContractId,
+		"rpcUrl":            e2eConfig.TargetNetworkRPCURL,
+		"account":           e2eConfig.TargetNetworkPublicKey,
+		"secretKey":         e2eConfig.TargetNetworkSecretKey,
+		"networkPassphrase": e2eConfig.TargetNetworkPassPhrase,
+		"functionName":      functionName,
+		"param1":            param1,
+	})
+	if err != nil {
+		return "", errors.Wrap(err, "failed to execute javascript template")
+	}
+
+	envCmd := cmd.NewCmd("node")
+	status, stdOutLines, err := e2e.RunCommandWithStdin(envCmd, e2eConfig, stdin)
+
+	if status != 0 || err != nil {
+		return "", fmt.Errorf("nodejs invoke of example contract %s had error %v, %v", contractName, status, err)
```

Comment:
same thing here, `stdOutLines` would be nice to see even on error

Reply:
this type of log behavior is tied into `VerboseOutput=true` param, to trigger stdout/err to print during `RunCommand` regardless, but it can get pretty noisy, maybe logging strategy needs this finer tuning in a follow-on tech debt cleanup or needs now?